### ISSUE TITLE
v3: `podSpecPatch` can result in failed workflow

### ISSUE THREAD

**Issue by alexec** [argoproj/argo-workflows#5897]
```
Pod "deploy-pipeline-kq79f-807066115" is invalid: [spec.containers[0].env[0].valueFrom: Invalid value: "": may not be specified when value is not empty, spec.containers[1].env[0].valueFrom: Invalid value: "": may not be specified when value is not empty
```

This workflow fails:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: hello-world-
  labels:
    workflows.argoproj.io/archive-strategy: "false"
spec:
  entrypoint: whalesay
  templates:
  - name: whalesay
    podSpecPatch: |
      {"containers":[{"name":"main","resources":{"limits":{"cpu":"500m","memory":"250Mi"}}},{"name":"wait","resources":{"limits":{"cpu":"500m","memory":"250Mi"}}}]}
    container:
      image: docker/whalesay:latest
      command: [cowsay]
      args: ["hello world"]

```

**Comment by alexec** [comment]
You can fix this issue by changing the order of containers in `podSpecPatch`:

BAD:

```json
      {"containers":[{"name":"main","resources":{"limits":{"cpu":"500m","memory":"250Mi"}}},{"name":"wait","resources":{"limits":{"cpu":"500m","memory":"250Mi"}}}]}
```

GOOD:

```json
      {"containers":[{"name":"wait","resources":{"limits":{"cpu":"500m","memory":"250Mi"}}},{"name":"main","resources":{"limits":{"cpu":"500m","memory":"250Mi"}}}]}
```

**Comment by alexec** [comment]
Could have been introduced by:

fix(controller): Fix podSpecPatch (#5360)

**Comment by alexec** [comment]
Created an upstream issue. https://github.com/golang/go/issues/46146
